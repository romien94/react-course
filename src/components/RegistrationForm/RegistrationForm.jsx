import React from "react";
import PropTypes from "prop-types";
import { Field, reduxForm } from "redux-form";
import { connect } from "react-redux";
import { registerUser } from "../../modules/actions";
import {
  required,
  email,
  minLength8,
  maxLength12,
} from "../../utils/validators";

import FormLabel from "../common/FormLabel";
import Input from "../common/Input";
import Button from "../common/Button";
import { Link } from "react-router-dom";

class RegistrationForm extends React.Component {
  static propTypes = {
    authenticate: PropTypes.func,
  };

  state = {
    email: "",
    password: "",
    firstName: "",
    secondName: "",
  };

  render() {
    return (
      <form
        className="app-form"
        onSubmit={this.props.handleSubmit((val) => {
          const { email, password, firstName, secondName } = val
          this.props.registerUser(email, password, firstName, secondName);
        })}
      >
        <div className="app-form__wrapper">
          <h2 className="app-form__title">Регистрация</h2>
          <p className="app-form__suggestion">
            Уже зарегистрированы?
            <Link to="/login">Войти</Link>
          </p>
          <div className="app-form__fields">
            <div className="app-form__row">
              <FormLabel>
                <span className="app-form__fieldname">
                  Адрес электронной почты
                </span>
                <Field
                  data-testid="email"
                  type="email"
                  className="app-form__input"
                  component={Input}
                  name="email"
                  validate={[required, email]}
                />
              </FormLabel>
            </div>
            <div className="app-form__row">
              <FormLabel>
                <span className="app-form__fieldname">Имя</span>
                <Field
                  data-testid="name"
                  type="text"
                  className="app-form__input"
                  component={Input}
                  name="name"
                  validate={[required]}
                />
              </FormLabel>
              <FormLabel>
                <span className="app-form__fieldname">Фамилия</span>
                <Field
                  data-testid="surname"
                  type="text"
                  className="app-form__input"
                  component={Input}
                  name="surname"
                  validate={[required]}
                />
              </FormLabel>
            </div>
            <div className="app-form__row">
              <FormLabel>
                <span className="app-form__fieldname">Пароль</span>
                <Field
                  data-testid="password"
                  type="text"
                  className="app-form__input"
                  component={Input}
                  name="password"
                  validate={[required, minLength8, maxLength12]}
                />
              </FormLabel>
            </div>
          </div>
          <div className="app-form__controls">
            <div className="app-form__row">
              <Button
                type="submit"
                data-testid="register"
                className="button app-form__button"
              >
                Зарегистрироваться
              </Button>
            </div>
          </div>
        </div>
      </form>
    );
  }
}

const RegistrationFormConnect = connect(
  (state) => ({ error: state.auth.error }),
  { registerUser }
)(RegistrationForm);

export default reduxForm({ form: "registrationForm" })(RegistrationFormConnect);
