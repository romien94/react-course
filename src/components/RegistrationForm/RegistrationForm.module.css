.appForm {
    position: relative;
    display: flex;
    flex-direction: column;
    flex: 1;
    max-width: 505px;
    min-height: 475px;
    background-color: white;
    padding: 65px 60px;
    color: #747474;
    font-size: 15px;
}

.appForm:before {
    position: absolute;
    top: 50%;
    right: 100%;
    content: '';
    display: block;
    background-image: url('../../assets/images/logo.png');
    background-repeat: no-repeat;
    width: 160px;
    height: 50px;
    transform: translate(-50%, -50%);
}


.formTitle {
    font-size: 36px;
    margin-bottom: 20px;
    color: black;
}

.formSuggestion {
    font-size: 16px;
    margin-bottom: 20px;
}

.formLink {
    color: #0074ea;
    text-decoration: none;
}

.formRow {
    display: flex;
    margin-bottom: 35px;
}

.formLabel {
    display: flex;
    flex-direction: column;
    flex: 1;
    margin-right: 15px;
}

.formLabel:last-child {
    margin-right: 0;
}

.formInput {
    border: none;
    background-color: transparent;
    border-bottom: 1px solid gray;
    padding: 7px 0;
    font-size: 15px;
}

.formLabel:hover .formInput {
    border-bottom-color: #0074ea;    
}

.formInput:active,
.formInput:focus {
    outline: none;
    border-bottom-color: #ffc700;
}


