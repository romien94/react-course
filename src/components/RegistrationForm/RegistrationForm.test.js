import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import RegistrationForm from "./RegistrationForm";

describe("Login works fine", () => {
  const { getByTestId } = render(<RegistrationForm />);
  it("Snapshot matches", () => {
    const mockRegistrationForm = renderer.create(<RegistrationForm />).toJSON();
    expect(mockRegistrationForm).toMatchSnapshot();
  });

  describe("Contains all the necessary inputs with basic setup", () => {
    expect(getByTestId("email")).toBeTruthy();
    expect(getByTestId("name")).toBeTruthy();
    expect(getByTestId("surname")).toBeTruthy();
    expect(getByTestId("password")).toBeTruthy();
    expect(getByTestId("register")).toBeTruthy();
    expect(getByTestId("register")).toHaveAttribute('type','submit');
  });
});
