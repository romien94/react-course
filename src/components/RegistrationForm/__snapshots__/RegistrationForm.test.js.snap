// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login works fine Snapshot matches 1`] = `
<form
  className="app-form"
>
  <div
    className="app-form__wrapper"
  >
    <h2
      className="app-form__title"
    >
      Регистрация
    </h2>
    <p
      className="app-form__suggestion"
    >
      Уже зарегистрированы?
      <a
        className="app-form__link"
        href=""
      >
        Войти
      </a>
    </p>
    <div
      className="app-form__fields"
    >
      <div
        className="app-form__row"
      >
        <label
          className="MuiFormLabel-root"
        >
          <span
            className="app-form__fieldname"
          >
            Адрес электронной почты
          </span>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline app-form__input"
            data-testid="email"
            onClick={[Function]}
          >
            <input
              className="MuiInputBase-input MuiInput-input"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="email"
            />
          </div>
        </label>
      </div>
      <div
        className="app-form__row"
      >
        <label
          className="MuiFormLabel-root"
        >
          <span
            className="app-form__fieldname"
          >
            Имя
          </span>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline app-form__input"
            data-testid="name"
            onClick={[Function]}
          >
            <input
              className="MuiInputBase-input MuiInput-input"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="text"
            />
          </div>
        </label>
        <label
          className="MuiFormLabel-root"
        >
          <span
            className="app-form__fieldname"
          >
            Фамилия
          </span>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline app-form__input"
            data-testid="surname"
            onClick={[Function]}
          >
            <input
              className="MuiInputBase-input MuiInput-input"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="text"
            />
          </div>
        </label>
      </div>
      <div
        className="app-form__row"
      >
        <label
          className="MuiFormLabel-root"
        >
          <span
            className="app-form__fieldname"
          >
            Пароль
          </span>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline app-form__input"
            data-testid="password"
            onClick={[Function]}
          >
            <input
              className="MuiInputBase-input MuiInput-input"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="text"
            />
          </div>
        </label>
      </div>
    </div>
    <div
      className="app-form__controls"
    >
      <div
        className="app-form__row"
      >
        <button
          className="button app-form__button"
          data-testid="register"
          type="submit"
        >
          Зарегистрироваться
        </button>
      </div>
    </div>
  </div>
</form>
`;
